internal/services/dataprotection/data_protection_backup_vault_resource.go

```diff
@@ -92,6 +100,16 @@ func resourceDataProtectionBackupVault() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice(backupvaults.PossibleValuesForSoftDeleteState(), false),
 			},
 
+			// NOTE O+C: we do not want to disable immutability if a user has enabled it via the portal or other methods.
+			"immutability": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(backupvaults.PossibleValuesForImmutabilityState(), false),
```

Comment:
We should have a CustomizeDiff that is ForceNew when changing this from `Locked` to something else (ie. Unlocked/Disabled)